### PR TITLE
gnrc: various fixes

### PR DESCRIPTION
**Pull request by OlegHahm**
* I don't see a reason for adding a link local address for every non-link local address. Linux also doesn't do so.
* 6LoWPAN routers were bit overzealous with detecting and removing false duplicates.

### CODE REVIEW

File: sys/net/gnrc/network_layer/sixlowpan/nd/gnrc_sixlowpan_nd.c [hunk 1]
```diff
@@ -291,7 +292,8 @@ uint8_t gnrc_sixlowpan_nd_opt_ar_handle(kernel_pid_t iface, ipv6_hdr_t *ipv6, ui
             if ((nc_entry != NULL) &&
                 ((gnrc_ipv6_nc_get_type(nc_entry) == GNRC_IPV6_NC_TYPE_REGISTERED) ||
                  (gnrc_ipv6_nc_get_type(nc_entry) == GNRC_IPV6_NC_TYPE_TENTATIVE)) &&
-                (ar_opt->eui64.uint64.u64 != nc_entry->eui64.uint64.u64)) {
+                ((nc_entry->eui64.uint64.u64 != 0) &&
```
Comment (authmillenon) [inline sys/net/gnrc/network_layer/sixlowpan/nd/gnrc_sixlowpan_nd.c]: How would the EUI64 field of an AR be legally 0?

Comment (OlegHahm) [inline sys/net/gnrc/network_layer/sixlowpan/nd/gnrc_sixlowpan_nd.c]: I'm checking the neighbor cache entry and this gets only filled by processing a neighbor advertisement. So, it's basically always 0 when a NCE already exist when you receive a NBR SOL for the first time.

Comment (authmillenon) [inline sys/net/gnrc/network_layer/sixlowpan/nd/gnrc_sixlowpan_nd.c]: Ah, right...

**Comment by authmillenon** [comment]
ACK if Travis is happy

**Comment by authmillenon** [comment]
Unittests are failing.

**Comment by OlegHahm** [comment]
I saw it. I'm on it.

**Comment by OlegHahm** [comment]
fixed

**Comment by authmillenon** [comment]
Now they are crashing. Also, I would prefer to rename the unittests changing commit to something more suitable for master ;).

**Comment by OlegHahm** [comment]
> Also, I would prefer to rename the unittests changing commit to something more suitable for master ;). 

You mean you want an extra commit?

**Comment by authmillenon** [comment]
Yepp

**Comment by OlegHahm** [comment]
Ok, will fix, if you're okay to squash.

The last commit should fix the unittests, although I don't really understand how the problem was related to the PR. It was crashing in `gnrc_ipv6_nc.c:228`, because the interface couldn't be found, because it was never added.

**Comment by authmillenon** [comment]
How is this file even in here?

```
Building application "unittests" for "arduino-mega2560" with MCU "atmega2560".
/home/travis/build/RIOT-OS/RIOT/sys/net/gnrc/network_layer/icmpv6/gnrc_icmpv6.c: In function 'gnrc_icmpv6_calc_csum':
/home/travis/build/RIOT-OS/RIOT/sys/net/gnrc/network_layer/icmpv6/gnrc_icmpv6.c:179:17: error: 'EBADMSG' undeclared (first use in this function)
         return -EBADMSG;
                 ^
```

A solution to fix that might be #3710.

**Comment by authmillenon** [comment]
#3710 was merged, please rebase

**Comment by authmillenon** [comment]
(and squash while your at it ;-))

**Comment by OlegHahm** [comment]
Double green.

**Pull request merged**